### ISSUE TITLE
Intent-to-Remove: loader allowlist

### ISSUE THREAD

**Issue by alanorozco** [ampproject/amphtml#33543]
We show loading indicators only for a select group of elements:

https://github.com/ampproject/amphtml/blob/246b09c4f8c80a621289d3b9dcb67436af94a291/src/layout.js#L96-L118

This list gets shipped as part of [`v0.js`](https://cdn.ampproject.org/v0.js). Introducing a new component with a loader **will then increase the size of `v0.js`** roughly by the size of its name, e.g. `'AMP-TIKTOK': !0`. 

## This proposes removing the allowlist by tagname

-   At one point, loaders were changed so that they would [only show up after 600ms of loading time](https://github.com/ampproject/amphtml/blob/212f4fe289e351f22090b10c5eb7508375c97a2e/extensions/amp-loader/0.1/amp-loader.js#L31-L33). **Loaders appear less frequently than when the list was introduced.**

-   The list is not widely known among contributors, so it's likely very out of date, and would further increase the size of v0.js if updated. **Switching to a simpler ruleset would prevent v0.js from growing further**, and would ensure that components get the same treatment when new.

### Replacement rule

Loaders will now be rendered any time a component's layout is **not** `container` **nor** `nodisplay`.

🔵  **includes**:

- `amp-img`
- `amp-video`
- `amp-iframe`
- `amp-list`
- (and more, see validation rules)

 ⛔️ **excludes**:

- `amp-story`
- `amp-sidebar`
- `amp-accordion`
- `amp-fit-text`
- (and more, see validation rules)

---

cc @ampproject/wg-components 
cc @ampproject/wg-performance

**Comment by newmuis** [comment]
I think this would make stories show a loader, where that would be undesirable.  How can we avoid this?

**Comment by newmuis** [comment]
I think this would make stories show a loader, where that would be undesirable. How can we avoid this?

**Comment by alanorozco** [comment]
@newmuis `amp-story` is excluded since it's `layout="container"`.

**Comment by caroqliu** [comment]
Does this change exclude certain components like `amp-fit-text`, etc. because the 600ms loading delay is not expected to be met, or for some other reason? I understand `container` and `nodisplay` layouts are still strictly excluded, however others that may not need a loader can be used with other layout types, such as `amp-fit-text` with `responsive`.

**Comment by alanorozco** [comment]
>  because the 600ms loading delay is not expected to be met, or for some other reason?

Yes, `amp-fit-text` doesn't actually load anything—its `layoutCallback` resolves after a mutation. 

>  others that may not need a loader can be used with other layout types, such as amp-fit-text with responsive.

Good point. I think I have to explore which components can be both size-defined and `container`, though I expect those to be on the same non-loading camp as `amp-fit-text`.

**Comment by caroqliu** [comment]
> > because the 600ms loading delay is not expected to be met, or for some other reason?
> 
> Yes, `amp-fit-text` doesn't actually load anything—its `layoutCallback` resolves after a mutation.
> 
👍 
> > others that may not need a loader can be used with other layout types, such as amp-fit-text with responsive.
> 
> Good point. I think I have to explore which components can be both size-defined and `container`, though I expect those to be on the same non-loading camp as `amp-fit-text`.
In that case I'd be more comfortable making this change once we have a better understanding of which components might be affected (or more confidence of which definitely won't be.)
